### PR TITLE
Add Propagators

### PR DESCRIPTION
**Pull request by szabo137**
This PR adds a simple implementation of fermion and boson propagators. 

This should be merged *before* #18 because the latter will use the functionality. 

TODOS: 

- [x] replace dev-version of `QEDbase` with release `v0.1.5` (cf. https://github.com/QEDjl-project/QEDbase.jl/issues/39)

### CODE REVIEW

**Comment by szabo137** [comment]
@AntonReinhard This should be ready for review now. Could you please have a look on this?

File: src/propagators.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+end
+
+function _fermion_propagator(P::QEDbase.AbstractFourMomentum)
+    return (slashed(P)) * _scalar_propagator(P)
+end
+
+function propagator(particle_type::BosonLike, K::QEDbase.AbstractFourMomentum)
+    return _scalar_propagator(K, mass(particle_type))
+end
+propagator(particle_type::Photon, K::QEDbase.AbstractFourMomentum) = _scalar_propagator(K)
```
Comment (AntonReinhard) [inline src/propagators.jl:57]: Can we make this into a whole `function ... end` too just for the niceness?

File: test/propagators.jl [hunk 2]
```diff
@@ -0,0 +1,26 @@
+using Random
+using QEDbase
+using QEDprocesses
+
+RNG = MersenneTwister(137137)
+ATOL = 0.0
```
Comment (AntonReinhard) [inline test/propagators.jl:6]: atol defaults to zero anyways

**Review by AntonReinhard — CHANGES_REQUESTED** [review]

File: test/propagators.jl (hunk 2 above)
Comment (szabo137) [reply test/propagators.jl:6]: This was made explicitly to not rely on defaults of `isapprox`, and it is better to read.

File: src/propagators.jl (hunk 1 above)
Comment (szabo137) [reply src/propagators.jl]: Niceness reestablished in 77c5494

Comment (AntonReinhard) [reply src/propagators.jl]: Thank you ^^

**Review by AntonReinhard — APPROVED** [review]

**Pull request merged**